quipucords/scanner/acs/inspect.py

```diff
@@ -0,0 +1,157 @@
+
+        inspection_status = SystemInspectionResult.SUCCESS
+        self.save_results(inspection_status, results)
+
+        if self.scan_task.systems_scanned:
+            return self.success_message, ScanTask.COMPLETED
+        return self.failure_message, ScanTask.FAILED
+
+    def _fetch_acs_metric(self, endpoint):
+        """Fetch ACS metrics data from a given endpoint."""
+        response = self.client.get(endpoint)
+        if not response.ok:
+            raise NotImplementedError()
+
+        return response.json()
+
+    def get_secured_units_current(self):
+        """Retrieve ACS secured node/cpu units metrics."""
+        secured_units_current = self._fetch_acs_metric(
+            "/v1/product/usage/secured-units/current"
+        )
+        return secured_units_current
+
+    def get_secured_units_max(self):
+        """Retrieve ACS max secured node/cpu units metrics."""
+        secured_units_max = self._fetch_acs_metric(
+            "/v1/product/usage/secured-units/max"
+        )
+        return secured_units_max
+
+    @transaction.atomic
+    def save_results(self, inspection_status, facts_dict: dict):
+        """
+        Save inspection results to the scan task.
+
+        Stores the results of the inspection in the system and increments the statistics
+        for the scan task.
+
+        :param inspection_status: The status of the inspection
+        :param facts_dict: Dictionary of facts to be saved to
+        """
+        system = self._persist_facts(inspection_status, facts_dict)
+        increment_kwargs = self._get_increment_kwargs(system.status)
+        self.scan_task.increment_stats(self.system_name, **increment_kwargs)
+
+    def _persist_facts(
+        self, inspection_status, facts_dict: dict
+    ) -> SystemInspectionResult:
+        """
+        Persist facts to database.
+
+        :param inspection_status: status of the inspection
+        :param facts_dict: dictionary of facts to persist
+        """
+        sys_result = SystemInspectionResult(
+            name=self.system_name,
+            status=inspection_status,
+            source=self.scan_task.source,
+            task_inspection_result=self.scan_task.inspection_result,
+        )
+        sys_result.save()
+        raw_facts = self._facts_dict_as_raw_facts(sys_result, **facts_dict)
+        RawFact.objects.bulk_create(raw_facts)
+        return sys_result
+
+    def _facts_dict_as_raw_facts(
+        self, inspection_result: SystemInspectionResult, **facts_dict
+    ) -> list[RawFact]:
+        """
+        Convert a dictionary of facts to a list of RawFacts.
+
+        :param inspection_result: The SystemInspectionResult that will be used to
+         determine if facts are valid
+        """
+        raw_facts = []
+        for fact_name, fact_value in facts_dict.items():
+            raw_facts.append(
+                RawFact(
+                    name=fact_name,
+                    value=json.dumps(fact_value, default=str),
```

Comment:
no json dumps, here. `RawFact.value` is a proper JsonField nowadays.
```suggestion
                    value=fact_value,
```